### ISSUE TITLE
path $..*[?(@ === false)] does not work as expected

### ISSUE THREAD

**Issue by vk342** [JSONPath-Plus/JSONPath#136]
given an object

{"a":{"b":false}}

I expect "$..*[?(@ === false)]" to return a result, in fact, this search returns empty array

As far as I can tell, the issue is on line 674:

```
JSONPath.prototype._eval = function (
    code, _v, _vname, path, parent, parentPropName
) {
    if (!this._obj || !_v) { return false; }

```
because _v === false the expression nether evaluated

**Comment by michelnev** [comment]
I see similar issues with numeric values of zero.

**Comment by brettz9** [comment]
Fix released in 5.0.4. Note that this change may now impact other uses of `@` where it is not a scalar such as:

`$..*[?(@ && @.prop === 'abc')]`

Previously, without the `@ && `, it would just silently fail to match, but now (as per other similar cases) it will throw.

@michelnev : You need to use `@property` lower case, but otherwise, the fix should address your case as well.
